src/2_RTB/NormeDiProgetto/content.typ

```diff
@@ -346,34 +377,163 @@ In ogni diagramma dei casi d'uso possono essere definite:
+                        ),
+                        caption: "Figura rappresentante la relazione di associazione."
+                )
+
+        - *Generalizzazione*:  una classe B eredita attributi, comportamenti e relazioni dalla classe genitore A. In altre parole, ogni istanza di B è anche un'istanza di A (subtyping), ma con specificità o dettagli aggiuntivi propri di B.
+
+Nella rappresentazione grafica, si utilizza una freccia vuota che punta dalla classe figlia (Classe B) alla classe genitore (Classe A).
+                #figure(
+                        image(
+                                "assets/imgs/Generalizzazione.png",
+                                height:38%
+                        ),
+                        caption: "Figura rappresentante la relazione di generalizzazione."
+                )
+
+        
+
+
 === Codifica 
 
 ==== Descrizione e scopo 
+
 L'attività di codifica viene svolta dai Programmatori, i quali sono responsabili della traduzione delle decisioni progettuali nel codice sorgente.
-I Programmatori seguono le linea guida e le _best practices_ stabilite durante la fase di progettazione architetturale.
+I Programmatori operano all'interno di un contesto ben definito seguendo le linee guida stabilite durante la fase di progettazione architetturale. Questo approccio garantisce coerenza nell'implementazione del design e nell'applicazione delle #glossary[best practices], favorendo la creazione di codice robusto, manutenibile e di alta qualità.
+
+Nel perseguire gli obiettivi di qualità, i Programmatori sono tenuti a rispettare le metriche definite nel _Piano di Qualifica v1.0_.
 
 ==== Aspettative 
+
+La fase di codifica è intrinsecamente orientata alla creazione di un prodotto software che soddisfi pienamente le esigenze e le aspettative della Proponente.
 Ci si aspetta che il codice sviluppato rispetti determinate caratteristiche:
 
         - Conformità alle specifiche;
         - Chiarezza e comprensibilità;
         - Ottimizzazione delle prestazioni;
         - Supplemento di test per verificare la correttezza e il funzionamento.
 
-==== Stile di codifica 
-[...]
+==== Stile di codifica   
+
+Il team #team ha deciso di adottare il linguaggio di programmazione Python per la creazione dei simulatori di sensori.
+Questa scelta tecnologica è motivata da diverse considerazioni che conferiscono a Python un ruolo privilegiato nel contesto di sviluppo.
+La familiarità che molti membri del team hanno con Python è un elemento chiave. La mancanza di una curva di apprendimento ripida consente al team di concentrarsi direttamente sulla realizzazione del prodotto, risparmiando una quantità significativa di tempo che altrimenti sarebbe stato utilizzato per apprendere un nuovo linguaggio.
+Inoltre, Python promuove uno stile di codifica uniforme e leggibile. La sua struttura rigida basata sull'indentazione agevola la comprensione della struttura del programma, contribuendo a una maggiore chiarezza all'interno del codice. 
```

Comment:
La frase sulla curva di apprendimento non mi piace in realtà: o la toglierei oppure ne cambiaerei la struttura. 
Tralatro non sono troppo sicuro di "Inoltre, Python promuove uno stile di codifica uniforme e leggibile. La sua struttura rigida basata sull'indentazione agevola la comprensione della struttura del programma, contribuendo a una maggiore chiarezza all'interno del codice"

```suggestion
Il team #team ha deciso di adottare il linguaggio di programmazione Python per la creazione dei simulatori di sensori.
Questa scelta tecnologica è motivata dalla familiarità che molti membri del team hanno con il linguaggio in questione, ciò consente al team di concentrarsi direttamente sulla realizzazione del prodotto, risparmiando una quantità significativa risorse. 
Inoltre, Python promuove uno stile di codifica uniforme e leggibile. La sua struttura rigida basata sull'indentazione agevola la comprensione della struttura del programma, contribuendo a una maggiore chiarezza all'interno del codice. 
```